src/IceRpc.Protobuf.Generators/Internal/ServiceClass.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) ZeroC, Inc.
+
+namespace IceRpc.Protobuf.Generators.Internal;
+
+/// <summary>Represents a C# class that has the <c>IceRpc.Protobuf.ProtobufServiceAttribute</c> and for which this generator
+/// implements the <c>IceRpc.IDispatcher</c> interface.</summary>
+internal class ServiceClass : ContainerDefinition
+{
+    /// <summary>Gets a value indicating whether the service has a base service class.</summary>
+    internal bool HasBaseServiceClass { get; }
```

Comment:
Protobuf services don't support any inheritance, so I don't understand how this would work.

I think for both Protobuf and Slice, it should be an error to apply the attribute to a class that does not implement any INameService. For example:

```csharp
[ProtobufService] internal partial class Foo {} // error
```

Reply:
This would not solve anything. It is possible for a C# class B to implement IBService and inherit from class A that implement IAService even is IBService and IAService are totally unrelated.